test/dataSources.spec.js

```diff
@@ -0,0 +1,154 @@
+/* eslint-env jest */
+
+const fs = require('fs');
+const path = require('path');
+const dataSources = require('../lib/dataSources');
+const { ConnectionFactory } = require('./setup-database');
+
+const dataSourceMetadata = fs.readFileSync(
+  path.resolve(`${__dirname}/fixtures/data_source_metadata.ttl`),
+  'utf8'
+);
+
+describe('data_sources', () => {
+  let conn;
+  const aDSName = 'MyDataSource';
+  const aOptions = {
+    'jdbc.driver': 'com.mysql.jdbc.Driver',
+    'jdbc.url': 'jdbc:mysql://localhost:3306/sys',
+    'jdbc.username': 'root',
+    'jdbc.password': '',
+    'ext.serverTimezone': 'EST',
+  };
+
+  beforeEach(() => {
+    conn = ConnectionFactory();
+  });
+
+  const assureExists = () =>
+    dataSources.list(conn).then(res => {
+      const exists =
+        res.body.data_sources && res.body.data_sources.includes(aDSName);
+      if (!exists) {
+        return dataSources.add(conn, aDSName, aOptions);
+      }
+      return Promise.resolve({ status: 201 });
+    });
+
+  const assureNotExists = () =>
+    dataSources.list(conn).then(res => {
+      const exists =
+        res.body.data_sources && res.body.data_sources.includes(aDSName);
+      if (exists) {
+        return dataSources.remove(conn, aDSName);
+      }
+      return Promise.resolve({ status: 204 });
```

Comment:
A little nitpicky, but for these two methods, why not just `return res` at the end? That way we don't have to put an additional Promise in the flow here, and we don't have to ensure that our hard-coded numbers always match the API's status codes.

Reply:
ah, sure